prov/verbs/src/verbs_info.c

```diff
@@ -377,54 +386,89 @@ static inline int fi_ibv_get_qp_cap(struct ibv_context *ctx,
 	return ret;
 }
 
-static int fi_ibv_get_device_attrs(struct ibv_context *ctx, struct fi_info *info)
+static int fi_ibv_get_device_attrs(struct ibv_context *ctx,
+				   struct fi_info *info,
+				   struct fi_ibv_device_info *device_info)
 {
-	struct ibv_device_attr device_attr;
-	struct ibv_port_attr port_attr;
+	struct ibv_device_attr *device_attr;
+	struct ibv_port_attr *port_attr;
 	int ret = 0;
 
-	ret = ibv_query_device(ctx, &device_attr);
-	if (ret) {
-		VERBS_INFO_ERRNO(FI_LOG_FABRIC, "ibv_query_device", errno);
-		return -errno;
+	if (!device_info->device_attr) {
+		device_attr = calloc(1, sizeof(*device_attr));
+		if (!device_attr)
+			return -FI_ENOMEM;
+		ret = ibv_query_device(ctx, device_attr);
+		if (ret) {
+			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
+					 "ibv_query_device", errno);
+			free(device_attr);
+			return -errno;
+		}
+		device_info->device_attr = device_attr;
+	} else {
+		device_attr = device_info->device_attr;
 	}
 
-	info->domain_attr->cq_cnt 		= device_attr.max_cq;
-	info->domain_attr->ep_cnt 		= device_attr.max_qp;
-	info->domain_attr->tx_ctx_cnt 		= MIN(info->domain_attr->tx_ctx_cnt, device_attr.max_qp);
-	info->domain_attr->rx_ctx_cnt 		= MIN(info->domain_attr->rx_ctx_cnt, device_attr.max_qp);
-	info->domain_attr->max_ep_tx_ctx 	= MIN(info->domain_attr->tx_ctx_cnt, device_attr.max_qp);
-	info->domain_attr->max_ep_rx_ctx 	= MIN(info->domain_attr->rx_ctx_cnt, device_attr.max_qp);
-	info->domain_attr->max_ep_srx_ctx	= device_attr.max_qp;
-	info->domain_attr->mr_cnt		= device_attr.max_mr;
+	info->domain_attr->cq_cnt 		= device_attr->max_cq;
+	info->domain_attr->ep_cnt 		= device_attr->max_qp;
+	info->domain_attr->tx_ctx_cnt 		= MIN(info->domain_attr->tx_ctx_cnt,
+						      device_attr->max_qp);
+	info->domain_attr->rx_ctx_cnt 		= MIN(info->domain_attr->rx_ctx_cnt,
+						      device_attr->max_qp);
+	info->domain_attr->max_ep_tx_ctx 	= MIN(info->domain_attr->tx_ctx_cnt,
+						      device_attr->max_qp);
+	info->domain_attr->max_ep_rx_ctx 	= MIN(info->domain_attr->rx_ctx_cnt,
+						      device_attr->max_qp);
+	info->domain_attr->max_ep_srx_ctx	= device_attr->max_srq;
```

Comment:
This indicates support for SRQs.  Carrying a flag seems unnecessary, or I'm not understanding its need.

Reply:
Thanks. Will re-write solution